src/tools/perf/perftest.c

```diff
@@ -384,28 +390,27 @@ static ucs_status_t setup_sock_rte_p2p(struct perftest_context *ctx)
                 connfd          = accept(sockfd, (struct sockaddr*)&client_addr,
                                          &client_addr_len);
                 if (connfd < 0) {
-                    ucs_error("accept() failed: %m");
+                    snprintf(err_str, 64, "accept() failed: %m");
                     status = UCS_ERR_IO_ERROR;
                     goto err_close_sockfd;
                 }
 
                 ucs_sockaddr_str((struct sockaddr*)&client_addr, addr_str,
                                  sizeof(addr_str));
                 printf("Accepted connection from %s\n", addr_str);
-                close(sockfd);
+                ucs_close_fd(&sockfd);
                 break;
             }
+
             snprintf(err_str, 64, "bind() failed: %m");
         }
-        close(sockfd);
-        sockfd = -1;
+
+        ucs_close_fd(&sockfd);
     }
 
-    if (sockfd < 0) {
-        ucs_error("%s failed. %s",
-                  (ctx->server_addr != NULL) ? "client" : "server", err_str);
+    if (connfd < 0) {
         status = UCS_ERR_IO_ERROR;
-        goto out_free_res;
+        goto err_close_sockfd;
```

Comment:
seems we can get here when "res" list is empty, for example, in which case err_str will not be initialized when jumping to err_close_sockfd and in line 478